### ISSUE TITLE
select_add_order_by: unnecessary error when target is hashed or sorted

### ISSUE THREAD

**Issue by mbtools** [abaplint/abaplint#1955]
```abap
REPORT zfoobar.
DATA itab TYPE HASHED TABLE OF string.
SELECT text FROM t100 INTO TABLE @itab.
DATA itab2 TYPE SORTED TABLE OF string.
SELECT text FROM t100 INTO TABLE @itab2.
```

/zfoobar.prog.abap[3, 1]: Add ORDER BY(select_add_order_by)
/zfoobar.prog.abap[5, 1]: Add ORDER BY(select_add_order_by)

**Comment by larshp** [comment]
@mbtools try again, 2.74.17
